src/applications/appeals/shared/validations/date.js

```diff
@@ -9,26 +13,80 @@ export const buildErrorParts = (month, day, year, maxDays) => {
   };
 };
 
-export const createPartsError = errorParts => {
-  return Object.entries(errorParts).reduce(
+// add second error message containing the part of the date with an error;
+// used to add `aria-invalid` to the specific input
+export const createScreenReaderErrorMsg = datePartErrors => {
+  return Object.entries(datePartErrors).reduce(
     (result, [partName, hasError]) => result + (hasError ? `${partName} ` : ''),
     '',
   );
 };
 
-export const isInvalidDateString = (year, day, month, dateString) => {
+export const isDateStringValid = (year, day, month, dateString) => {
```

Comment:
Let's stick with `isInvalidDateString`. `isDateStringValid` implies that the string is valid if this function returns `true`, which isn't the case.